src/main/java/Deadline.java

```diff
@@ -0,0 +1,37 @@
+public class Deadline extends Task {
+    protected String deadline;
+
+    public Deadline(String description, String deadline) {
+        super(description);
+        this.deadline = deadline;
+        type = TaskType.DEADLINE;
```

Comment:
Perhaps you can align with the above statement and put `this.type` instead of just `type`. This applies for the other task subclasses as well.

Reply:
I was trying to follow the code guidelines for this! There is a clause on this: "Avoid unnecessary use of this with fields."

For `deadline` there is a conflicting name in the method parameter and as part of the object, so the explicit `this` keyword was necessary. However, for the `type` field there was no such ambiguity so I went ahead and omitted `this`!